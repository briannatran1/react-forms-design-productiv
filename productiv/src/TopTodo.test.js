import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

describe('TopTodo component', function () {
  const todos = [{
    id: 1,
    title: 'testing',
    description: 'finish testing',
    priority: 2
  },
  {
    id: 2,
    title: 'eat',
    description: 'eat lunch',
    priority: 1
  }
  ];

  test('renders without crashing', function () {
    render(<TopTodo todos={todos} />);
  });

  test('matches snapshot', function () {
    const { asFragment } = render(<TopTodo todos={todos} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('contains only highest priority todo', function () {
    const { queryByText, container } = render(<TopTodo todos={todos} />);

    expect(queryByText('eat')).toBeInTheDocument();
    expect(queryByText('(Priority: 1)')).toBeInTheDocument();
    expect(queryByText('eat lunch')).toBeInTheDocument();

    expect(container.querySelectorAll('.Todo').length).toEqual(1);

    expect(queryByText('testing')).not.toBeInTheDocument();
    expect(queryByText('finish testing')).not.toBeInTheDocument();
    expect(queryByText('(Priority: 2)')).not.toBeInTheDocument();
  });
});