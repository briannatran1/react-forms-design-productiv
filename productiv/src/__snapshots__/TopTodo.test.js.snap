// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div>
      <b>
        eat
      </b>
       
      <small>
        (Priority: 1)
      </small>
    </div>
    <div>
      <small>
        eat lunch
      </small>
    </div>
  </div>
</DocumentFragment>
`;
